src/zoid/checkout/component.jsx

```diff
@@ -317,6 +317,11 @@ export function getCheckoutComponent(): CheckoutComponent {
           queryParam: true,
           required: false,
         },
+
+        hostedButtonId: {
+          type: "string",
+          required: false,
+        },
```

Comment:
does the checkout ui need this?

Reply:
yes but only for the pre-render button click https://github.com/paypal/paypal-checkout-components/pull/2276/files#diff-f15dc9fa288e403c9d09722ceae61adb63327f473aa7807fbf3cb49896732f7cR81-R86

Reply:
what is checkout doing with it?

Reply:
SPB destructures the `hostedButtonId` from `prerenderDetails` here - https://github.com/paypal/paypal-smart-payment-buttons/blob/c268be3d0b18ff0ecaa11df646e74c8787ec03ff/src/button/button.js#L393-L397

and then adds it into `payment` here - https://github.com/paypal/paypal-smart-payment-buttons/blob/c268be3d0b18ff0ecaa11df646e74c8787ec03ff/src/button/button.js#L413-L419

then initiate payment is called right below - https://github.com/paypal/paypal-smart-payment-buttons/blob/c268be3d0b18ff0ecaa11df646e74c8787ec03ff/src/button/button.js#L420

and then passed in here when `Checkout({` is invoked - https://github.com/paypal/paypal-smart-payment-buttons/blob/c268be3d0b18ff0ecaa11df646e74c8787ec03ff/src/payment-flows/checkout.js#L330

this took me a significant amount of time to figure out so I'm super open to removing it (because of the increased complexity) and/or identifying another solution - wdyt?

Reply:
we chatted offline about this, checkout is not actually using this value, going to remove it from zoid